src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowTransformerControlTest.java

```diff
@@ -0,0 +1,442 @@
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+class AcLoadFlowTransformerControlTest {
+
+    private Network network;
+    private Bus bus1;
+    private Bus bus2;
+    private Bus bus3;
+    private Bus bus4;
+    private Line line12;
+    private Line line24;
+    private TwoWindingsTransformer t2wt;
+    private ThreeWindingsTransformer t3wt;
+    private Load load3;
+
+    private LoadFlow.Runner loadFlowRunner;
+    private LoadFlowParameters parameters;
+
+    @BeforeEach
+    void setUp() {
+        loadFlowRunner = new LoadFlow.Runner(new OpenLoadFlowProvider(new DenseMatrixFactory()));
+        parameters = new LoadFlowParameters();
+        parameters.setTransformerVoltageControlOn(false);
+        parameters.setDistributedSlack(false);
+        OpenLoadFlowParameters parametersExt = new OpenLoadFlowParameters()
+                .setSlackBusSelector(new FirstSlackBusSelector());
+        this.parameters.addExtension(OpenLoadFlowParameters.class, parametersExt);
+    }
+
+    @Test
+    void baseCaseT2wtTest() {
+        selectNetwork(createNetworkWithT2wt());
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(134.273, bus2);
+        assertVoltageEquals(27.0038, bus3);
+    }
+
+    @Test
+    void tapPlusTwoT2wtTest() {
+        selectNetwork(createNetworkWithT2wt());
+
+        t2wt.getRatioTapChanger().setTapPosition(2);
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(134.281, bus2);
+        assertVoltageEquals(34.427, bus3);
+    }
+
+    @Test
+    void voltageControlT2wtTest() {
+        selectNetwork(createNetworkWithT2wt());
+
+        parameters.setTransformerVoltageControlOn(true);
+        t2wt.getRatioTapChanger()
+                .setTargetDeadband(0)
+                .setRegulating(true)
+                .setTapPosition(0)
+                .setRegulationTerminal(t2wt.getTerminal2())
+                .setTargetV(34.0);
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+        assertVoltageEquals(134.281, bus2);
+        assertVoltageEquals(34.433, t2wt.getTerminal2().getBusView().getBus());
+        assertEquals(2, t2wt.getRatioTapChanger().getTapPosition());
+    }
+
+    @Test
+    void remoteVoltageControlT2wtTest() {
+        selectNetwork(createNetworkWithT2wt());
+
+        parameters.setTransformerVoltageControlOn(true);
+        t2wt.getRatioTapChanger()
+                .setTargetDeadband(0)
+                .setRegulating(true)
+                .setTapPosition(0)
+                .setRegulationTerminal(t2wt.getTerminal1())
+                .setTargetV(134.0);
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
```

Comment:
@geofjamg it is not working: here we have a RTC on a T2WT that is trying to regulate on side 1. But we also have a generator that maintains voltage at 135.0kV. What I don't understand is: why we are in step zero but voltages are not equals to base test case. I have maybe a problem of initialization after the firsyt iteration of the outerloop.